sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/BooleanLiteralUnnecessaryBase.cs

```diff
@@ -0,0 +1,286 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2019 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class BooleanLiteralUnnecessaryBase<TBinaryExpression, TSyntaxKind> : SonarDiagnosticAnalyzer
+        where TBinaryExpression : SyntaxNode
+        where TSyntaxKind : struct
+    {
+        protected enum ErrorLocation
+        {
+            // The BooleanLiteral node is highlighted
+            Normal,
+
+            // The BooleanLiteral node and the preceding operator are highlighted together
+            Extended,
+
+            // Inside a binary expression, the non-boolean literal is highlighted
+            // e.g. for 'a && True', 'a' will be highlighted
+            Inverted
+        }
+
+        protected internal const string DiagnosticId = "S1125";
+        protected const string MessageFormat = "Remove the unnecessary Boolean literal(s).";
+
+        protected abstract DiagnosticDescriptor Rule { get; }
```

Comment:
No need for that, use `SupportedDiagnostics[0]` instead of `Rule` in the diagnostic creation.